bazel/genrule_repository.bzl

```diff
@@ -84,12 +84,24 @@ def _absolute_bin(path):
 def _genrule_environment(ctx):
   lines = []
 
+  # Bazel uses the same command for C and C++ compilation.
+  c_compiler = ctx.var['CC']
+
   # Bare minimum cflags to get included test binaries to link.
   #
   # See //tools:bazel.rc for the full set.
   asan_flags = ["-fsanitize=address,undefined"]
   tsan_flags = ["-fsanitize=thread"]
 
+  # Required when using the sanitizer modes in older GCC versions. This is
+  # not a great solution because it doesn't detect GCC when Bazel has wrapped
+  # it in an intermediate script, but it works well enough to keep CI running.
```

Comment:
Can you describe the versions involved? We can't bump CI to a newer version and set that as the floor?

Reply:
@mattklein123 reported errors with `/usr/bin/g++-5`, and the linked StackOverflow post indicates it affects some unknown version range in (5.x, 6.x).

Upgrading CI GCC to v7 would probably fix this but I don't want to start doing wild stuff until the build is more settled.

Reply:
OK, please just add to comment. Thanks.

Reply:
Done.